lib/data/multipleBackendGateway.js

```diff
@@ -287,7 +302,18 @@ const multipleBackendGateway = {
                         dataStoreType: client.clientType,
                         dataStoreVersionId,
                     };
-                    cb(err, dataRetrievalInfo);
+                    if (err) {
+                        // if error copying obj, counter should be decremented
+                        locationStorageCheck(destLocationConstraintName,
```

Comment:
This looks fine per se, but to avoid surprises I would think it's better to map exactly what is done for the other rollback cases, i.e. do `return locationStorageCheck` here to make it synchronous with the response, and `return cb(null, dataRetrievalInfo)` on success.